src/components/BoardPiece.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import CanvasImage from './CanvasImage';
+import { Element } from '../types';
+import { IconMenu, MenuItem, IconButton } from 'material-ui';
+import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
+
+interface Props {
+  height: number;
+  width: number;
+  x: number;
+  y: number;
+  src: string;
+  element: Element;
+}
+
+interface State {
+  value: string[];
+}
+
+class BoardPiece extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      value: []
+    };
+  }
+
+  handleChange(value: any | any[]) {
+    this.setState({ value: value });
```

Comment:
This will requires some firebase interactions further down the lane to notify everyone of the state. So, it will better to keep the current state of piece in the store itself, other way, you can just dispatch an action with value for updating which then calls firebase with specific values.